content/guides/typescript-for-solid.mdx

```diff
@@ -0,0 +1,709 @@
+This is usually not an issue, but in asynchronous contexts or during [`onCleanup`](/reference/oncleanup), it may occur.
+
+<Aside>
+If you find the above method a bit shorthand, there's a more explicit alternative. 
+Solid's [`<Show>`](/reference/show) component actually performs memoization, so you don't necessarily need a memo for this:
+
+```tsx
+const user = createMemo(() => {
+  const v = data()
+  return v && v.type === "user" && v // Type: User | undefined
+})
+
+<Show when={user()}>
+  {user => <User user={user()} />}
+</Show>
+```
+While this approach is more explicit, it's worth noting that the primary example for the callback form should ideally not use this technique for the sake of clarity
+</Aside>
+
+## Advanced JSX attributes and directives
+
+### Custom event handlers
+
+For custom events in Solid, the [`on:___`]() and [`oncapture:___`]() attributes can be utilized. 
+To type these events, you'll need to extend Solid's JSX namespace.
+
+```tsx
+class NameEvent extends CustomEvent {
+  type: "Name";
+  detail: { name: string };
+
+  constructor(name: string) {
+    super("Name", { detail: { name } });
+  }
+}
+
+declare module "solid-js" {
+  namespace JSX {
+    interface CustomEvents {
+      Name: NameEvent;  // Matches `on:Name`
+    }
+    interface CustomCaptureEvents {
+      Name: NameEvent;  // Matches `oncapture:Name`
+    }
+  }
+}
+
+// Usage
+<div on:Name={(event) => console.log("name is", event.detail.name)} />;
+```
+
+<Aside>
+**Note**: By default, using native events like `mousemove` with the `on` prefix (e.g., <div on:mousemove={e => {}} />) will result in a TypeScript error. 
+This is because they are not included in the custom event typings for Solid.
+To solve this, extend the `CustomEvents` and `CustomCaptureEvents` interfaces to include events from the `HTMLElementEventMap`:
+
+To include all native events:
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface CustomEvents extends HTMLElementEventMap {}
+    interface CustomCaptureEvents extends HTMLElementEventMap {}
+  }
+}
+```
+
+To include specific native events, you can pick certain events (e.g. `mousemove` and `pointermove`):
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface CustomEvents extends Pick<HTMLElementEventMap, "mousemove" | "pointermove"> {}
+  }
+}
+```
+</Aside>
+
+#### Forcing properties and custom attributes 
+
+Solid's [`prop:___`]() allows forcing properties, while [`attr:___`]() permits custom attributes. 
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface ExplicitProperties {
+      count: number;
+      name: string;
+    }
+    interface ExplicitAttributes {
+      count: number;
+      name: string;
+    }
+  }
+}
+
+// Usage
+<Input prop:name={name()} prop:count={count()}/>
+<my-web-component attr:name={name()} attr:count={count()}/>
+```
+
+#### Custom directives
+
+Custom directives can be applied using the [`use:___`]() attribute.
```

Comment:
I added this @otonashixav based on a comment you made. Do you think this is an appropriate location for it based on what you were talking about?